docs/docs/recipes.md

```diff
@@ -770,6 +770,124 @@ export const pageQuery = graphql`
+```
+
+Add the following plugin entry to your `gatsby-config.js` file. Use your own space ID and access token from the API keys section of your Contentful space. You may also use [environment variables](https://www.gatsbyjs.org/docs/environment-variables/) to store your space and token.
+
+```javascript:title=gatsby-config.js
+plugins: [
+  {
+    resolve: `gatsby-source-contentful`,
+    options: {
+      spaceId: `your_space_id`,
+      accessToken: `your_access_token`,
+    },
+  },
+]
+```
+
+Run `gatsby develop` and make sure the site compiled successfully.
+
+##### Querying your data
+
+Open the [Gatsby GraphiQL editor](https://www.gatsbyjs.org/docs/introducing-graphiql/) by visiting `https://localhost:8000/___graphql`. The Contentful plugin adds several new node types to your site, including a node type for every content type in your Contentful website. For our example site with a `News` content type and two fields, we get an `allContentfulNews` node type in GraphQL.
+
+![the graphql interface, with a sample query outlined below](./images/recipe-sourcing-contentful-graphql.png)
+
+To query all our news article from Contentful, we can use the following GraphQL query. Note that the `body` field is an object. [Long-text field types in Contentful are returned as objects](https://www.gatsbyjs.org/packages/gatsby-source-contentful/#a-note-about-longtext-fields) instead strings.
+
+```graphql
+{
+  allContentfulNews {
+    edges {
+      node {
+        body {
+          body
+        }
+        title
+      }
+    }
+  }
+}
+```
+
+Contentful nodes also include several metadata fields like `createdAt` or `node_locale`.
+
+##### Using Contentful data in your site
+
+You can use Contentful data like any other data source. A simple page listing news articles from our sample space would look like this:
+
+```jsx
+import React from "react"
+import { graphql } from "gatsby"
+
+const NewsPage = ({ data }) => (
+  <div>
+    <h1>News</h1>
+    {data.allContentfulNews.edges.map(({ node }) => (
+      <>
+        <h2>{node.title}</h2>
+        <p>
+          <strong>{node.createdAt}</strong>
+          {" — "}
+          {node.body.body}
+        </p>
+      </>
+    ))}
+  </div>
+)
+
+export default NewsPage
+
+export const query = graphql`
+  {
+    allContentfulNews(sort: { fields: [updatedAt] }) {
+      edges {
+        node {
+          title
+          body {
+            body
+          }
+          createdAt(formatString: "MMMM DD YYYY")
+        }
+      }
+    }
+  }
+```
+
+#### Additional resources
+
+- [Building a Site with React and Contentful](https://www.gatsbyjs.org/blog/2018-1-25-building-a-site-with-react-and-contentful/)
+- [More on Sourcing from Contentful](https://www.gatsbyjs.org/docs/sourcing-from-contentful/)
+- [Contentful source plugin](https://www.gatsbyjs.org/packages/gatsby-source-contentful/)
+
+##### Live example
+
+- [Example website](http://kevee.net/gatsby-source-contentful-recipe/)
```

Comment:
Thx for contributing the demo. It is ok to have an example on an external server? Maybe we should host it on the Gatsby demo server?

Reply:
Oh yeah - the convention is to include them in the Gatsby repo. Here's a recipe that does it: https://www.gatsbyjs.org/docs/recipes/#sourcing-markdown-data-for-blog-posts-and-pages-with-graphql

Reply:
I can include the example in the repo, but now that I look at it, there isn't anything in the example repo that isn't already outlined in the recipe itself. Should I add the example, or remove references to the live example?

Reply:
I think we should add the example - it would help adoption and get people playing around with it! It makes sense for more in-depth recipes like this one - perhaps not for some of the smaller-scoped ones.

Reply:
Added example in https://github.com/gatsbyjs/gatsby/pull/17318/commits/9d341360d5eaa00ca341816a64551c6286a10c83. Also, the link in the recipe will of course be broken for now until the pull is merged.